### PR TITLE
Add javadocs in the HTTPRequestSessionContext methods

### PR DESCRIPTION
**Pull request by andreyliporage**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: ---

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked if the purpose of the task was reached with this PR
- [x] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [x] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [x] Checked that there are no performance issues created by these changes or additions
- [x] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [ ] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [ ] Checked if there is no possible merge conflicts before reviewing the PR
- [ ] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [ ] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [ ] Checked that code coverage for this PR is higher than 80%.
- [ ] Checked that there are no problems and code smells identified by the quality and security tools
- [ ] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [ ] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [ ] Checked that all unitary and integration tests ran by the CI system are passing
- [ ] Checked that there are no performance issues created by these changes or additions
- [ ] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [ ] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 1]
```diff
@@ -348,6 +357,18 @@ public boolean supportClientTransaction()
         return queryRequestMetadata;
     }
 
+    /**
+     * Returns a list of the HTTP headers received on the request.
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:361]: Redundant with @return

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 2]
```diff
@@ -358,11 +379,29 @@ public boolean supportClientTransaction()
...
+     *
+     * @return a Map with the {@link PRESTO_SESSION} properties
+     */
     private static Map<String, String> parseSessionHeaders(MultivaluedMap<String, String> headers)
     {
         return parseProperty(headers, PRESTO_SESSION);
     }
 
+    /**
+     * Checks if the received headers matches the defined headers pattern.
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:395]: This is only true for the PRESTO_ROLE header. The method is specific to this header.

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 3]
```diff
@@ -400,18 +458,48 @@ public boolean supportClientTransaction()
...
+     * @return a Set with {@code PRESTO_CLIENT_CAPABILITIES} headers.
+     */
     private static Set<String> parseClientCapabilities(MultivaluedMap<String, String> headers)
     {
         Splitter splitter = Splitter.on(',').trimResults().omitEmptyStrings();
         return ImmutableSet.copyOf(splitter.split(nullToEmpty(headers.getFirst(PRESTO_CLIENT_CAPABILITIES))));
     }
 
+    /**
+     * Check which resource will be started.
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:494]: - resource **estimate**;
- Extracts PRESTO_RESOURCE_ESTIMATE header;

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 4]
```diff
@@ -439,6 +527,16 @@ private static ResourceEstimates parseResourceEstimate(MultivaluedMap<String, St
         return builder.build();
     }
 
+    /**
+     * Sets a default for serializedData and transforms ir to json.
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:531]: - typo;
- this method extracts the PRESTO_QUERY_REQUEST_METADATA header, parses escaped slashes and double quotes, than deserialize to QueryRequestMetadata with a json codec.

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 5]
```diff
@@ -487,6 +607,17 @@ private static void assertRequest(boolean expression, String format, Object... a
         return preparedStatements.build();
     }
 
+    /**
+     * Checks if the parameter is null and instantiates a new {@link TransactionId}
```
Comment (caiopavanelli) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:611]: change **parameter** to **transactionId**

File: presto-main/src/main/java/io/prestosql/Session.java [hunk 6]
```diff
@@ -788,6 +788,13 @@ public ResourceEstimateBuilder setPeakMemory(DataSize peakMemory)
             return this;
         }
 
+        /**
+         * Estimated resource usage for a query.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/Session.java:792]: "Estimate the resource usage for an executed query."

File: presto-main/src/main/java/io/prestosql/Session.java [hunk 7]
```diff
@@ -788,6 +788,13 @@ public ResourceEstimateBuilder setPeakMemory(DataSize peakMemory)
...
+         * <p>
+         * Takes the runtime, cpu runtime and amount of memory used.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/Session.java:794]: "Takes the estimate of the CPU runtime and memory used for an executed query. "

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 8]
```diff
@@ -510,6 +642,17 @@ private static WebApplicationException badRequest(String message)
                 .build());
     }
 
+    /**
+     * Check and return a given string if it is non-null and non-empty.
+     * <p>
+     * Returns the given string if is non-null and non-empty.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:648]: I think we don't need this description.

Also, haven't you already documented a method similar this one in the PR before?

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 9]
```diff
@@ -487,6 +608,17 @@ private static void assertRequest(boolean expression, String format, Object... a
         return preparedStatements.build();
     }
 
+    /**
+     * Checks if the transactionId is null and instantiates a new {@link TransactionId}
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:612]: Try to instantiate a new TransactionId

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 10]
```diff
@@ -487,6 +608,17 @@ private static void assertRequest(boolean expression, String format, Object... a
...
+     * <p>
+     * Checks whether the transactionId is null. if so, an empty option is returned.
+     * If not, a new instance of the {@link TransactionId} is returned. For invalid values,
+     * an WebApplicationException is thrown and handled.
+     *
+     * @param transactionId an string that represents the transaction id.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:618]: "a string..."

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 11]
```diff
@@ -487,6 +608,17 @@ private static void assertRequest(boolean expression, String format, Object... a
...
+     *
+     * @return an Optional of type TransactionId
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:620]: an Optional TransactionId

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 12]
```diff
@@ -208,6 +208,15 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
         return identity;
     }
 
+    /**
+     * Creates a session identity.
+     *
+     * @param authenticatedIdentity an Optional of {@link Identity}
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:214]: an Optional Session {@link Identity}

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 13]
```diff
@@ -208,6 +208,15 @@ public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuth
...
+     * @param headers a MultivaluedMap containing all request headers.
+     * @param groupProvider a {@link GroupProvider} interface implementation
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:216]: a {@link GroupProvider} object

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 14]
```diff
@@ -358,11 +379,29 @@ public boolean supportClientTransaction()
                 .collect(toImmutableList());
     }
 
+    /**
+     * Extracts a specific header PRESTO_SESSION
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:383]: "Extracts the specific..."

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 15]
```diff
@@ -379,11 +418,30 @@ public boolean supportClientTransaction()
         return roles.build();
     }
 
+    /**
+     * Extracts a specific header PRESTO_EXTRA_CREDENTIAL
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:422]: "Extracts the..."

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 16]
```diff
@@ -400,18 +458,48 @@ public boolean supportClientTransaction()
...
     {
         Splitter splitter = Splitter.on(',').trimResults().omitEmptyStrings();
         return ImmutableSet.copyOf(splitter.split(nullToEmpty(headers.getFirst(PRESTO_CLIENT_TAGS))));
     }
 
+    /**
+     * Extracts the header PRESTO_CLIENT_CAPABILITIES.
+     * <p>
+     * Extract the header and add it to a Set.
+     * Returns a Set with the extracted {@code PRESTO_CLIENT_CAPABILITIES} headers.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:481]: ".... header."

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 17]
```diff
@@ -439,6 +527,17 @@ private static ResourceEstimates parseResourceEstimate(MultivaluedMap<String, St
         return builder.build();
     }
 
+    /**
+     * Extracts the Presto_QUERY_REQUEST_METADATA header, parses escaped slashes and double quotes, than
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:531]: This information about parsing is not necessary.
Only keeps: "Extracts the Presto_QUERY_REQUEST_METADATA header"

File: presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java [hunk 18]
```diff
@@ -453,13 +552,35 @@ private static ResourceEstimates parseResourceEstimate(MultivaluedMap<String, St
...
+     */
     private static void assertRequest(boolean expression, String format, Object... args)
     {
         if (!expression) {
             throw badRequest(format(format, args));
         }
     }
 
+    /**
+     * Extract the PRESTO_PREPARED_STATEMENT header and parser the sql command.
```
Comment (joaopedro-simbiose) [inline presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java:575]: "...and parse the sql command."
